.github/workflows/release.yml

```diff
@@ -0,0 +1,129 @@
+        required: true
+      developmentBranch:
+        description: 'Development branch to cut the release from'
+        default: development
+        required: true
+      stableBranch:
+        description: 'Stable branch to merge the development branch into'
+        default: stable
+        required: true
+      releaseBranch:
+        description: 'Release branch to create (e.g. 1.0.x for version 1.0.0; once created, branch protection rules apply)'
+        default: dry_run
+        required: true
+      nextVersion:
+        description: 'Next version after release (e.g. 1.1.0, -SNAPSHOT will be added automatically)'
+        required: true
+      nextMicroVersion:
+        description: 'Next version after release for release branch (e.g. 1.0.1, -SNAPSHOT will be added automatically)'
+        required: true
+jobs:
+  build:
+    env:
+      MAVEN_ARGS: "--no-transfer-progress --batch-mode"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the relevant timefold-solver tag
+        uses: actions/checkout@v3
+        with:
+          repository: "TimefoldAI/timefold-solver"
+          path: "./timefold-solver"
+          fetch-depth: 0
+          ref: v${{ github.event.inputs.version }}
+
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'maven'
+
+      # No need to wait for the upstream release to show up in Maven Central.
+      - name: Build the upstream release tag
+        working-directory: "./timefold-solver"
+        run: |
+          mvn -Dquickly install
+          cd ..
+          rm -rf timefold-solver
+
+      - name: Checkout timefold-quickstarts
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Create release branch and switch to it
+        run: |
+          git config user.name "Timefold Release Bot"
+          git config user.email "release@timefold.ai"
+          git checkout -B ${{ github.event.inputs.releaseBranch }}
+
+      # We do not use versions:set, because we'd have to have the SNAPSHOT version built from somewhere,
+      # and at this point in the release, there is no upstream branch anywhere that would have this version anymore.
+      - name: Set release version and build release
+        run: |
+          DETECTED_VERSION=$(mvn -q \
+            -Dexec.executable="echo" \
+            -Dexec.args='${project.version}' \
+            --non-recursive \
+            org.codehaus.mojo:exec-maven-plugin:1.6.0:exec \
+          )
+          find -name "pom.xml" | xargs sed -i "s/$DETECTED_VERSION/${{ github.event.inputs.version }}/g"
+          find -name "build.gradle" | xargs sed -i "s/$DETECTED_VERSION/${{ github.event.inputs.version }}/g"
+          mvn -Dfull test
+          find -name "pom.xml" | xargs git add
+          find -name "build.gradle" | xargs git add
+          git commit -m "chore: release version ${{ github.event.inputs.version }}"
```

Comment:
The version replacement deserves a standalone script; especially if we repeat the same steps for the micro version below.